raven/src/main/java/org/opencadc/raven/RavenInitAction.java

```diff
@@ -0,0 +1,234 @@
+
+    static final String JNDI_DATASOURCE = "jdbc/inventory"; // context.xml
+
+    static final String SCHEMA_KEY = RAVEN_KEY + ".inventory.schema";
+
+    static final String PUBKEYFILE_KEY = RAVEN_KEY + ".publicKeyFile";
+    static final String PRIVKEYFILE_KEY = RAVEN_KEY + ".privateKeyFile";
+    static final String READ_GRANTS_KEY = RAVEN_KEY + ".readGrantProvider";
+    static final String WRITE_GRANTS_KEY = RAVEN_KEY + ".writeGrantProvider";
+
+    static final String DEV_AUTH_ONLY_KEY = RAVEN_KEY + ".authenticateOnly";
+
+    // set init initConfig, used by subsequent init methods
+
+    MultiValuedProperties props;
+    private URI resourceID;
+    private Map<String,Object> daoConfig;
+
+    public RavenInitAction() {
+        super();
+    }
+
+    @Override
+    public void doInit() {
+        initConfig();
+        initStorageSite();
+    }
+
+    /**
+     * Read config file and verify that all required entries are present.
+     *
+     * @return MultiValuedProperties containing the application config
+     */
+    static MultiValuedProperties getConfig() {
+        PropertiesReader r = new PropertiesReader("raven.properties");
+        MultiValuedProperties mvp = r.getAllProperties();
+
+        StringBuilder sb = new StringBuilder();
+        sb.append("incomplete config: ");
+        boolean ok = true;
+
+        // validate required config here
+        String schema = mvp.getFirstPropertyValue(RavenInitAction.SCHEMA_KEY);
+        sb.append("\n\t").append(RavenInitAction.SCHEMA_KEY).append(": ");
+        if (schema == null) {
+            sb.append("MISSING");
+            ok = false;
+        } else {
+            sb.append("OK");
+        }
+
+        String pub = mvp.getFirstPropertyValue(RavenInitAction.PUBKEYFILE_KEY);
+        sb.append("\n\t").append(RavenInitAction.PUBKEYFILE_KEY).append(": ");
+        if (pub == null) {
+            sb.append("MISSING");
+            ok = false;
+        } else {
+            sb.append("OK");
+        }
+
+        String priv = mvp.getFirstPropertyValue(RavenInitAction.PRIVKEYFILE_KEY);
+        sb.append("\n\t").append(RavenInitAction.PRIVKEYFILE_KEY).append(": ");
+        if (priv == null) {
+            sb.append("MISSING");
+            ok = false;
+        } else {
+            sb.append("OK");
+        }
+
+        if (!ok) {
+            throw new IllegalStateException(sb.toString());
+        }
+
+        return mvp;
+    }
+
+
+    private void initConfig() {
+        log.info("initConfig: START");
+        this.props = getConfig();
+        String rid = props.getFirstPropertyValue(RESOURCE_ID_KEY);
+        
+        try {
+            this.resourceID = new URI(rid);
+            log.info("initConfig: OK");
+        } catch (URISyntaxException ex) {
+            throw new IllegalStateException("invalid config: " + RESOURCE_ID_KEY + " must be a valid URI");
+        }
+    }
+
+    private void initStorageSite() {
```

Comment:
raven is not a StorageSite definitely does not put  a StorageSite record into the database. The database pool props try to clearly state that this is a query pool (like the query pool in luskan) and thus is a read-only pool. You must have configured with the wrong account to make this work, but it doesn't belong here at all and the pool should use a read-only account (tapuser in the cadc-postgresql-dev container)